drag-and-drop.md

```diff
@@ -0,0 +1,126 @@
+# Drag-adn-Drop
+
+The proposal to provide native support Drag-and Drop opeerations:
+* drag and drop between widgets 
+* receive drag and drop URL(s) to application from a desktop or another application
+* start drag operation from Fyne application with URL(s) to another desltop application (pending external changes)
+
+## Background
+
+There are few issues opened and related to missing drag and drop functionality:
+* Add drag and drop support https://github.com/fyne-io/fyne/issues/142
+* Drag and drop to application icon on macOS https://github.com/fyne-io/fyne/issues/497
+
+Existing interface provide a minimum capabilities to programmatically start a drug operation and get notified when it complets.
+This functionality doesn't provide an interface to communicate between widgets within the application nor send/receive drag and drop with other applications.
+
+## Architecture / API
+
+Existing interface will not change to preserve compatibility with existing code.
+The proposed changes will extend the interface and give user missing capabilities.
+
+The **fyne.CanvasObject** wishing to initiate drag operation must implement **fyne.Draggable** interface. No interface changes.
```

Comment:
This is the one part of the design that trips me up. Draggable is designed to be use for CanvasObjects that can be dragged. In this case the implicit Payload is a `CanvasObject`. Perhaps it is OK, as if there is no `Info` implemented it provides the old behaviour. Not too sure about this... I will think more.

Reply:
by defining a new interface we make this optional and user can create they own interfaces if they want.

Reply:
Sorry I don't quite understand this response.
It is possible that we need to come back to this once the discussion about drag events is reviewed?